src/pages/dashboard/analysis/index.tsx

```diff
@@ -0,0 +1,221 @@
+import { Col, Dropdown, Menu, Row } from 'antd';
+import React, { Component, Suspense } from 'react';
+
+import { Dispatch } from 'redux';
+import { Icon as LegacyIcon } from '@ant-design/compatible';
+import { GridContent } from '@ant-design/pro-layout';
+import { RadioChangeEvent } from 'antd/es/radio';
+import { RangePickerValue } from 'antd/es/date-picker/interface';
+import { connect } from 'dva';
+import PageLoading from './components/PageLoading';
+import { getTimeDistance } from './utils/utils';
+import { AnalysisData } from './data.d';
+import styles from './style.less';
+
+const IntroduceRow = React.lazy(() => import('./components/IntroduceRow'));
+const SalesCard = React.lazy(() => import('./components/SalesCard'));
+const TopSearch = React.lazy(() => import('./components/TopSearch'));
+const ProportionSales = React.lazy(() => import('./components/ProportionSales'));
+const OfflineData = React.lazy(() => import('./components/OfflineData'));
+
+interface AnalysisProps {
+  dashboardAndanalysis: AnalysisData;
+  dispatch: Dispatch<any>;
+  loading: boolean;
+}
+
+interface AnalysisState {
+  salesType: 'all' | 'online' | 'stores';
+  currentTabKey: string;
+  rangePickerValue: RangePickerValue;
+}
+
+@connect(
+  ({
+    dashboardAndanalysis,
+    loading,
+  }: {
+    dashboardAndanalysis: any;
+    loading: {
+      effects: { [key: string]: boolean };
+    };
+  }) => ({
+    dashboardAndanalysis,
+    loading: loading.effects['dashboardAndanalysis/fetch'],
+  }),
+)
+class Analysis extends Component<AnalysisProps, AnalysisState> {
+  state: AnalysisState = {
+    salesType: 'all',
+    currentTabKey: '',
+    rangePickerValue: getTimeDistance('year'),
+  };
+
+  reqRef: number = 0;
+
+  timeoutId: number = 0;
+
+  componentDidMount() {
+    const { dispatch } = this.props;
+    this.reqRef = requestAnimationFrame(() => {
+      dispatch({
+        type: 'dashboardAndanalysis/fetch',
+      });
+    });
+  }
+
+  componentWillUnmount() {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [expected call-signature: 'componentWillUnmount' to have a typedef](https://app.codacy.com/app/afc163/ant-design-pro/pullRequest?prid=4583897)